components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIMgtAdminServiceImpl.java

```diff
@@ -480,4 +482,114 @@ public String getEndpointGatewayConfig(String endpointId) throws APIManagementEx
 
     }
 
+    @Override
+    public String addBlockCondition(BlockConditions blockConditions) throws APIManagementException {
+        try {
+            return policyDAO.addBlockConditions(blockConditions);
+        } catch (APIMgtDAOException e) {
+            String errorMessage = "Couldn't add block condition.";
+            log.error(errorMessage, e);
+            throw new APIManagementException(errorMessage, ExceptionCodes.APIMGT_DAO_EXCEPTION);
+        }
+    }
+
+    @Override
+    public boolean updateBlockConditionStateByUUID(String uuid, Boolean state) throws APIManagementException {
+        try {
+            return policyDAO.updateBlockConditionStateByUUID(uuid, state);
+        } catch (APIMgtDAOException e) {
+            String errorMessage = "Couldn't update block condition.";
```

Comment:
Shall we add the context information(uuid and state) to the error log. Check the whole PR, there are multiple places we need to add the context information relative to the context.